### PR TITLE
Difference in code highlight depending on url #5041

### LINKED ISSUE #5041: Difference in code highlight depending on url
**Describe the bug**

When viewing [my comment as part of a post](https://dev.to/nombrekeff/what-would-you-call-this-function-3kbg#ijn0) the code blocks look great:
image

but when using [the deeplink that the comment offers](https://dev.to/karfau/comment/ijn0):
image

or when previewing markdown while adding a comment:
image

it's really hard to spot the code blocks.

**To Reproduce**

1. Example string to add to comment box:
```
Let's check what `code` looks like in a preview on the post page
```
2. Click the "preview" button

**Expected behavior**

Code blocks look the same everywhere, hopefully the way they look in posts.

**Desktop (please complete the following information):**

- OS: Manjaro
- Browser: Chromium
- Version: 78.0.3904.97 (Official Build) Arch Linux (64-bit)

### PR DESCRIPTION
**Pull request by ChintuKarthi**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Related Tickets & Documents

Closes #5041

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## Rubocop
While checking the Rubocop I came across three issues. So fixed them along with this PR. Attaching the screenshots for reference.

Before:
image

After:
image

## [optional] What gif best describes this PR or how it makes you feel?

happy gif

### CODE REVIEW

**Comment by ChintuKarthi** [comment]
> There's also the possibility that `<code>` is part of a `<pre>` and we might want to update that as well, right now it looks like:
> 
> Screenshot_2020-04-07 this is some code def f print( data ) regular text this is the pr - DEV(local)

@rhymes  How should the code within the pre tag should look like? Is there any background color preferred?

**Comment by rhymes** [comment]
@ChintuKarthi as preference I'm unsure, probably just a lighter dark grey is fine? 

cc. @ludwiczakpawel might have a better idea

**Comment by ChintuKarthi** [comment]
> @ChintuKarthi as preference I'm unsure, probably just a lighter dark grey is fine?
> 
> cc. @ludwiczakpawel might have a better idea

@rhymes  @ludwiczakpawel 

Adding a standard rgba(0, 0, 0, 0.1) is not having a much effect since we already have a pre tag that has a black background as shown in the below screenshot.

image

What I am thinking is, having a white background inside the black background of the pre tag might give a better user experience. Like the below-attached screenshot.

image

Or do you want me to follow the same standard of showing the ruby code inside pre tag just like how the python code is handled?

**Comment by ChintuKarthi** [comment]
@rhymes @ludwiczakpawel  any update on above-requested doubt?

**Comment by rhymes** [comment]
Hi @ChintuKarthi, I don't know if there's a standard for that, I think your suggestions is fine :-)

**Comment by ChintuKarthi** [comment]
> Hi @ChintuKarthi, I don't know if there's a standard for that, I think your suggestion is fine :-)

Thanks @rhymes  I'll follow the same.

**Comment by ChintuKarthi** [comment]
> Thanks for the changes @ChintuKarthi!
> 
> Although now it works identically no matter which is the page that render the comment, there might be too much margin around the pre block now:
> 
> Screenshot_2020-04-23 Discussion of The Moving Toyshop Fugiat ratione - DEV(local)(1)
> 
> you can reproduce it locally by creating a comment with the following content: https://gist.githubusercontent.com/rhymes/39a672188378ebb68b965a075ff27129/raw/ac78c057bad6c3745d9bef68e090feeeca4e91d8/gistfile1.txt
> 
> ps. I rebased your branch as it was based on an older version of master with a Ruby version we haven't used in a while

@rhymes  Could you please explain what does the too much margin here means. Is it the space around the code tag?

**Comment by rhymes** [comment]
> @rhymes Could you please explain what does the too much margin here means. Is it the space around the code tag?

yes, that space you see in white in the screenshot

**Comment by ChintuKarthi** [comment]
> > @rhymes Could you please explain what does the too much margin here means. Is it the space around the code tag?
> 
> yes, that space you see in white in the screenshot

Thanks, @rhymes I am able reduce the spacing around the code tag. Below is the screenshot. Please let me know if this is ok.

Single Comment Page:

image

Full Discussion Page:

image

When I tried to commit the code, I was facing some eslint issues in the existing code. I am working on it. I fixed around 26 Rubocop which was there. I am quite new to eslint. So I'll try to fix it ASAP and push the commit to this PR. Thanks.

**Comment by ChintuKarthi** [comment]
@rhymes Could you help me with some reviewers, please. This PR has been waiting here for a long time. Thanks.

File: db/schema.rb [hunk 1]
```diff
@@ -308,7 +308,6 @@
     t.string "slug"
     t.string "status", default: "active"
     t.datetime "updated_at", null: false
-    t.index ["slug"], name: "index_chat_channels_on_slug", unique: true
```
Comment (ludwiczakpawel) [inline db/schema.rb:311]: Just to be sure: should that be part of this PR? cc @rhymes

File: app/assets/stylesheets/comments.scss [hunk 2]
```diff
@@ -851,9 +873,13 @@ a.header-link {
       overflow-x: auto;
       code {
         background: #29292e;
-        color: #eff0f9;
+        color: black;
         white-space: pre;
         font-size: 1em;
+        background-color: white;
+        padding-top: 0%;
```
Comment (ludwiczakpawel) [inline app/assets/stylesheets/comments.scss:880]: Skip the `%` here since its not needed.

Comment (ChintuKarthi) [reply app/assets/stylesheets/comments.scss]: Removed `%` in [this](https://github.com/thepracticaldev/dev.to/pull/7103/commits/af9233a10f446ca71a71b00fbabb4e7531de2e5a#diff-53e600e3c9b5ce6a6226b7245ec587f3R885) commit.

File: db/schema.rb (hunk 1 above)
Comment (ChintuKarthi) [reply db/schema.rb]: This was removed when I ran bin/setup command. So does [this](https://github.com/thepracticaldev/dev.to/pull/7103/commits/af9233a10f446ca71a71b00fbabb4e7531de2e5a#diff-1acd2e7e27a227829d5d14a91c863bb6R1221). If these aren't needed please let me know I can push another commit to revert them.

File: db/schema.rb [hunk 3]
```diff
@@ -1216,9 +1216,9 @@
     t.datetime "updated_at", null: false
```
Comment (rhymes) [inline db/schema.rb:1216]: this file should not be part of this PR, please remove it.

Comment (ChintuKarthi) [reply db/schema.rb]: Removed it in [this](https://github.com/thepracticaldev/dev.to/pull/7103/commits/bea5bdd2efec42579c18e07618c604d33b7511a7#diff-1acd2e7e27a227829d5d14a91c863bb6L1216) commit.

Comment (ludwiczakpawel) [reply db/schema.rb]: I can still see some updates in that `schema.rb` file. I think you may need to just drop this file from your PR.

Comment (ChintuKarthi) [reply db/schema.rb]: @ludwiczakpawel  reverted the changes done in the schema file. Please verify now.

**Pull request merged**